test/e2e/tests/test_sagemaker_endpoint_autoscaling.py

```diff
@@ -0,0 +1,200 @@
+    )
+
+    yield (
+        model_input,
+        model_response,
+        endpoint_config_input,
+        endpoint_config_response,
+        endpoint_input,
+        endpoint_response,
+        resource_id,
+    )
+    sagemaker_client.delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client.delete_model(ModelName=model_name)
+
+@service_marker
+@pytest.mark.canary
+class TestSageMakerEndpointAutoscaling:
+    resource_id: str
+
+    def _generate_sagemaker_target_spec(self, bootstrap_resources: TestBootstrapResources):
+        target_resource_name = random_suffix_name("sagemaker-scalable-target", 32)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["SCALABLETARGET_NAME"] = target_resource_name
+        replacements["RESOURCE_ID"] = self.resource_id
+
+        spec_file = load_autoscaling_resource(
+            "sagemaker_endpoint_autoscaling_target", additional_replacements=replacements
+        )
+
+        # Create the k8s resource
+        reference = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, TARGET_RESOURCE_PLURAL, target_resource_name, namespace="default"
+        )
+
+        return (target_resource_name, reference, spec_file)
+
+    def _generate_sagemaker_policy_spec(self, bootstrap_resources: TestBootstrapResources):
+        policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["SCALINGPOLICY_NAME"] = policy_resource_name
+        replacements["RESOURCE_ID"] = self.resource_id
+        
+
+        spec_file = load_autoscaling_resource(
+            "sagemaker_endpoint_autoscaling_policy", additional_replacements=replacements
+        )
+
+        # Create the k8s resource
+        reference = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, POLICY_RESOURCE_PLURAL, policy_resource_name, namespace="default"
+        )
+
+        return (policy_resource_name, reference, spec_file)
+    
+    def _get_sagemaker_scalable_target_exists(self, applicationautoscaling_client, target_name: str) -> bool:
+        targets = applicationautoscaling_client.describe_scalable_targets(
+            ServiceNamespace="sagemaker",
+            ResourceIds=[
+                self.resource_id,
+                ],
+        )
+        
+        return len(targets["ScalableTargets"]) == 1
+
+    def _get_sagemaker_scaling_policy_exists(self, applicationautoscaling_client, policy_name: str) -> bool:
+        targets = applicationautoscaling_client.describe_scaling_policies(
+            ServiceNamespace="sagemaker",
+            ResourceId=self.resource_id,
+        )
+        
+        return len(targets["ScalingPolicies"]) == 1
+
+    def test_autoscaling(self, applicationautoscaling_client, sdk_endpoint):
+        self.resource_id = sdk_endpoint[6]
```

Comment:
`[6]` super worries me because if we modify that tuple at all, we're going to accidentally break a lot of things. I might recommend converting the tuple into a `NamedTuple` type from the `typing` package.

Eg:
```python
# At the top of the file

from typing import NamedTuple, Dict
class SDKEndpointReturnType(NamedTuple):
  model_input: Dict
  model_response: Dict
  endpoint_config_input: Dict
  endpoint_config_response: Dict
  endpoint_input: Dict
  endpoint_response: Dict
  resource_id: str
  
...
# Inside the `sdk_endpoint` fixture
return SDKEndpointReturnType(
   model_input,
   model_response,
   endpoint_config_input,
   endpoint_config_response,
   endpoint_input,
   endpoint_response,
   resource_id,
)

...
# Inside test_smoke
self.resource_id = sdk_endpoint.resource_id
```

Reply:
cleaning up this code bit as mentioned.